### PR TITLE
Add tutorial notebook for PEFT

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/11216?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11216/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11216
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Add a tutorial notebook for PEFT (QLoRA) with MLflow. The main docs will be updated in a separate PR.

**Note**: It seems the new linter confuses the shell script cell with Python. 

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by harupy — APPROVED** [review]
LGTM!

File: docs/source/llms/transformers/tutorials/fine-tuning/transformers-peft.ipynb [hunk 1]
```diff
@@ -0,0 +1,1281 @@
...
+   "metadata": {},
+   "source": [
+    "# Fine-Tuning Open-Source LLM using QLoRA with MLflow and PEFT"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "<p></p>\n",
```
Comment (B-Step62) [inline docs/source/llms/transformers/tutorials/fine-tuning/transformers-peft.ipynb:14]: Found that we can insert the download link in the notebook using "raw" cell, so we can remove the [workaround](http://localhost:8000/llms/transformers/index.html#download-the-use-case-tutorial-notebooks-to-try-them-locally) of putting the button in index page. If this sounds good, I will apply same to the other Transformers notebook in the follow-up cc: @BenWilson2

File: docs/source/llms/transformers/tutorials/fine-tuning/transformers-peft.ipynb [hunk 2]
```diff
@@ -0,0 +1,1281 @@
...
+     "title": ""
+    },
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "% pip install mlflow>=2.11.0\n",
+    "% pip install transformers peft accelerate bitsandbytes datasets -q -U"
```
Comment (harupy) [inline docs/source/llms/transformers/tutorials/fine-tuning/transformers-peft.ipynb:160]:
```suggestion
    "%pip install mlflow>=2.11.0\n",
    "%pip install transformers peft accelerate bitsandbytes datasets -q -U"
```

File: docs/source/llms/transformers/tutorials/fine-tuning/transformers-peft.ipynb [hunk 3]
```diff
@@ -0,0 +1,1281 @@
...
+      "|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |\r\n",
+      "|        ID   ID                                                             Usage      |\r\n",
+      "|=======================================================================================|\r\n",
+      "|  No running processes found                                                           |\r\n",
+      "+---------------------------------------------------------------------------------------+\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "% nvidia-smi"
```
Comment (harupy) [inline docs/source/llms/transformers/tutorials/fine-tuning/transformers-peft.ipynb:108]: Does Jupyter support this magic command?

Comment (harupy) [reply docs/source/llms/transformers/tutorials/fine-tuning/transformers-peft.ipynb:108]: What happens if I run-all this notebook and `nvidia-smi` is missing?

Comment (B-Step62) [reply docs/source/llms/transformers/tutorials/fine-tuning/transformers-peft.ipynb]: > Does Jupyter support this magic command?

Nope, fixed to `%sh nvidia-smi`.

> What happens if I run-all this notebook and nvidia-smi is not installed?

The first cell should fail with `command not found`, which is desired as we don't expect users to run this notebook without CUDA GPU backend.

File: docs/source/llms/transformers/tutorials/fine-tuning/transformers-peft.ipynb (hunk 1 above)
Comment (BenWilson2) [reply docs/source/llms/transformers/tutorials/fine-tuning/transformers-peft.ipynb:14]: No, I agree with you. Let's file a ticket for next sprint to migrate to the raw embed within notebook approach. It's cleaner. We should do this for all notebooks in all sections. Probably a 0.5 point ticket :)

**Pull request merged**